[UnifiedPDF] Fix coordinate space for accessibility.

ginalined opened the pull request:
#### 0d1d56d4da989bb82028ad08c8d392dc2b3ab51e
<pre>
[UnifiedPDF] Fix coordinate space for accessibility.

Reviewed by Abrar Rahman Protyasha.

- Fix VO cursor position does not match the focused element in unified pdf.
There are 2 parts of this change. One in pdfKit and one in webKit
- Update the plugin in WKAccessibilityDocumentObject to UnifiedPDFPlugin
- Remove extra functions
- Add new accessibility support api.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
(WebKit::PDFPluginBase::PDFPluginBase):
(WebKit::PDFPluginBase::convertFromPDFViewToRootView const): Deleted.
(WebKit::PDFPluginBase::convertFromPDFViewToScreenForAccessibility const): Deleted.
(WebKit::PDFPluginBase::convertFromRootViewToPDFView const): Deleted.
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::UnifiedPDFPlugin):
(WebKit::UnifiedPDFPlugin::accessibilityScrollToPage):
(WebKit::UnifiedPDFPlugin::convertFromPDFPageToScreenForAccessibility const):
(WebKit::UnifiedPDFPlugin::accessibilityHitTestIntPoint const):
* Source/WebKit/WebProcess/Plugins/PDF/WKAccessibilityPDFDocumentObject.h:
* Source/WebKit/WebProcess/Plugins/PDF/WKAccessibilityPDFDocumentObject.mm:
(-[WKAccessibilityPDFDocumentObject setPDFPlugin:]):
(-[WKAccessibilityPDFDocumentObject accessibilityFocusedUIElement]):
(-[WKAccessibilityPDFDocumentObject accessibilityWindow]):
(-[WKAccessibilityPDFDocumentObject accessibilityTopLevelUIElement]):
(-[WKAccessibilityPDFDocumentObject accessibilityVisibleChildren]):
(-[WKAccessibilityPDFDocumentObject accessibilitySubrole]):
(-[WKAccessibilityPDFDocumentObject accessibilityFrame]):
(-[WKAccessibilityPDFDocumentObject accessibilityParent]):
(-[WKAccessibilityPDFDocumentObject accessibilityAttributeValue:]):
(-[WKAccessibilityPDFDocumentObject accessibilityArrayAttributeCount:]):
(-[WKAccessibilityPDFDocumentObject accessibilityChildren]):
(-[WKAccessibilityPDFDocumentObject convertFromPDFPageToScreenForAccessibility:pageIndex:]):
(-[WKAccessibilityPDFDocumentObject accessibilityHitTest:]):
(-[WKAccessibilityPDFDocumentObject gotoDestination:]):
(-[WKAccessibilityPDFDocumentObject parent]): Deleted.

[UnifiedPDF] Fix coordinate space for accessibility <a href="https://bugs.webkit.org/show_bug.cgi?id=269956">https://bugs.webkit.org/show_bug.cgi?id=269956</a> <a href="https://rdar.apple.com/123459892">rdar://123459892</a>

Reviewed by Abrar Rahman Protyasha.

- Fix VO cursor position does not match the focused element in unified pdf.
There are 2 parts of this change. One in pdfKit and one in webKit
- Update the plugin in WKAccessibilityDocumentObject to UnifiedPDFPlugin
- Remove extra functions
- Add new accessibility support api

Canonical link: <a href="https://commits.webkit.org/275848@main">https://commits.webkit.org/275848@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9c906a4714ea74b24b6547c8ffe2fb5584ba43f1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43009 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22035 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/45411 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/45633 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/39134 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/25760 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/19458 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/35564 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/43582 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/19069 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/37033 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/16559 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/16661 "Passed tests") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1065 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/39192 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/38422 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/47173 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/17856 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/14712 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/42349 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/19461 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/41004 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9586 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/19640 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/19091 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
```diff
@@ -120,6 +120,11 @@ class UnifiedPDFPlugin final : public PDFPluginBase, public WebCore::GraphicsLay
 
+#if PLATFORM(MAC)
+    WebCore::FloatRect convertFromPDFViewToScreenForAccessibility(const WebCore::FloatRect&, int) const;
+    void accessibilityScrollToPage(int);
```
Use `PDFDocumentLayout::PageIndex` instead of `int`

smfr commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
```diff
@@ -120,6 +120,11 @@ class UnifiedPDFPlugin final : public PDFPluginBase, public WebCore::GraphicsLay
     float documentFittingScale() const { return m_documentLayout.scale(); }
 
+#if PLATFORM(MAC)
+    WebCore::FloatRect convertFromPDFViewToScreenForAccessibility(const WebCore::FloatRect&, int) const;
```
It's not clear what the last argument is. In the implementation it looks like a `PDFDocumentLayout::PageIndex`, in which case this function should be called `convertFromPDFPageToScreenForAccessibility()`

smfr commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
```diff
@@ -2823,10 +2829,37 @@ static NSStringCompareOptions compareOptionsForFindOptions(WebCore::FindOptions
+
+FloatRect UnifiedPDFPlugin::convertFromPDFViewToScreenForAccessibility(const FloatRect& rect, int pageIndex) const
+{
+    auto rectInPluginCoordinates = convertUp(CoordinateSpace::PDFPage, CoordinateSpace::Plugin, rect, pageIndex);
```
You should not call `convertUp()` off the main thread.

smfr commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
```diff
@@ -2823,10 +2829,37 @@ static NSStringCompareOptions compareOptionsForFindOptions(WebCore::FindOptions
 {
-    auto convertedPoint =  convertFromRootViewToPDFView(point);
-    return [m_accessibilityDocumentObject accessibilityHitTest:convertedPoint];
+    WebCore::FloatPoint floatPoint = FloatPoint(point);
```
The canonical way to write this is `auto floatPoint = FloatPoint { point }`. No need for `WebCore::` namespacing in the cpp file.

smfr commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
```diff
@@ -2823,10 +2829,37 @@ static NSStringCompareOptions compareOptionsForFindOptions(WebCore::FindOptions
-    auto convertedPoint =  convertFromRootViewToPDFView(point);
-    return [m_accessibilityDocumentObject accessibilityHitTest:convertedPoint];
+    WebCore::FloatPoint floatPoint = FloatPoint(point);
+    WebCore::FloatPoint pointInDocumentSpace = convertDown(CoordinateSpace::Plugin, CoordinateSpace::PDFDocumentLayout, floatPoint);
```
`auto`

smfr commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
```diff
@@ -2823,10 +2829,37 @@ static NSStringCompareOptions compareOptionsForFindOptions(WebCore::FindOptions
+    if (!hitPageIndex || *hitPageIndex >= m_documentLayout.pageCount())
+        return { };
+    auto pageIndex = *hitPageIndex;
+    RetainPtr page = m_documentLayout.pageAtIndex(pageIndex);
```
This is unused.

smfr commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
```diff
@@ -2823,10 +2829,37 @@ static NSStringCompareOptions compareOptionsForFindOptions(WebCore::FindOptions
+
+    CGPoint CGpointInPDFPageSpace = CGPointMake(pointInPDFPageSpace.x(), pointInPDFPageSpace.y());
+
+    return [m_accessibilityDocumentObject accessibilityHitTest:CGpointInPDFPageSpace];
```
There is automatic conversion from FloatPoint to CGPoint, so you can just pass `pointInPDFPageSpace` here.

smfr commented on Source/WebKit/WebProcess/Plugins/PDF/WKAccessibilityPDFDocumentObject.mm:
```diff
@@ -231,12 +273,10 @@ - (void)gotoDestination:(PDFDestination *)destination
 
-    callOnMainRunLoopAndWait([protectedPlugin = plugin, &point] {
-        protectedPlugin->setScrollOffset(point);
+    callOnMainRunLoopAndWait([protectedPlugin = plugin, &pageIndex] {
```
Is the "and wait" really necessary for triggering a scroll?

smfr commented on Tools/MiniBrowser/MiniBrowser.xcodeproj/xcshareddata/xcschemes/MiniBrowser.xcscheme:
```diff
@@ -6,6 +6,20 @@
       parallelizeBuildables = "YES"
       buildImplicitDependencies = "NO">
       <BuildActionEntries>
+         <BuildActionEntry
```
You should not include this change in the PR.

cdumez commented on Source/WebKit/WebProcess/Plugins/PDF/WKAccessibilityPDFDocumentObject.mm:
```diff
@@ -85,12 +86,28 @@ - (id)accessibilityFocusedUIElement
 
+- (id)accessibilityWindow
+{
+#pragma GCC diagnostic push
```
We have macros for this in WebKit:
```
ALLOW_DEPRECATED_DECLARATIONS_BEGIN
    doDeprecatedStuff;
ALLOW_DEPRECATED_DECLARATIONS_END
```

cdumez commented on Source/WebKit/WebProcess/Plugins/PDF/WKAccessibilityPDFDocumentObject.mm:
```diff
@@ -85,12 +86,28 @@ - (id)accessibilityFocusedUIElement
+- (id)accessibilityTopLevelUIElement
+{
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```
ditto.

cdumez commented on Source/WebKit/WebProcess/Plugins/PDF/WKAccessibilityPDFDocumentObject.mm:
```diff
@@ -101,13 +118,29 @@ - (NSArray *)accessibilityVisibleChildren
+- (NSRect)accessibilityFrame
+{
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```
ditto.

cdumez commented on Source/WebKit/WebProcess/Plugins/PDF/WKAccessibilityPDFDocumentObject.mm:
```diff
@@ -214,11 +247,20 @@ - (NSArray*)accessibilityChildren
+{
+    RefPtr plugin = _pdfPlugin.get();
+    WebCore::FloatRect rect = (WebCore::FloatRect)bound;
+    if (plugin)
```
```
if (RefPtr plugin = _pdfPlugin.get())
    return plugin->convertFromPDFPageToScreenForAccessibility(bound, pageIndex);
return bound;
```

cdumez commented on Source/WebKit/WebProcess/Plugins/PDF/WKAccessibilityPDFDocumentObject.mm:
```diff
@@ -214,11 +247,20 @@ - (NSArray*)accessibilityChildren
 
+- (NSRect)convertFromPDFPageToScreenForAccessibility:(NSRect)bound pageIndex:(WebKit::PDFDocumentLayout::PageIndex)pageIndex
+{
+    RefPtr plugin = _pdfPlugin.get();
```
We can reduce the scope, see comment below.

cdumez commented on Source/WebKit/WebProcess/Plugins/PDF/WKAccessibilityPDFDocumentObject.mm:
```diff
@@ -231,12 +273,10 @@ - (void)gotoDestination:(PDFDestination *)destination
 
-    callOnMainRunLoopAndWait([protectedPlugin = plugin, &point] {
-        protectedPlugin->setScrollOffset(point);
+    callOnMainRunLoop([protectedPlugin = plugin, &pageIndex] {
```
Now that you're no longer calling `AndWait()`, you can no longer capture stuff by reference. It isn't safe.
`&pageIndex` -> `pageIndex`
also, `protectedPlugin = plugin` -> `plugin`. No reason for the more complicated code.

aprotyas commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
```diff
@@ -2866,10 +2872,36 @@ static NSStringCompareOptions compareOptionsForFindOptions(WebCore::FindOptions
+
+FloatRect UnifiedPDFPlugin::convertFromPDFPageToScreenForAccessibility(const FloatRect& rect, PDFDocumentLayout::PageIndex pageIndex) const
+{
+    return WebCore::Accessibility::retrieveValueFromMainThread<WebCore::FloatRect>([&] () -> WebCore::FloatRect {
```
You can drop the `WebCore::` namespace prefix

aprotyas replied:
Also, our style guide says no space between the capture list and the arg list

```suggestion
    return WebCore::Accessibility::retrieveValueFromMainThread<FloatRect>([&]() ->FloatRect {
```

aprotyas commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
```diff
@@ -2866,10 +2872,36 @@ static NSStringCompareOptions compareOptionsForFindOptions(WebCore::FindOptions
+    scrollToPage(pageIndex);
+}
+
+FloatRect UnifiedPDFPlugin::convertFromPDFPageToScreenForAccessibility(const FloatRect& rect, PDFDocumentLayout::PageIndex pageIndex) const
```
Consider a different name for the `rect` argument, one which reflects which coordinate space it's in?

aprotyas commented on Source/WebKit/WebProcess/Plugins/PDF/WKAccessibilityPDFDocumentObject.mm:
```diff
@@ -214,11 +239,19 @@ - (NSArray*)accessibilityChildren
+    WebCore::FloatRect rect = (WebCore::FloatRect)bound;
+    if (RefPtr plugin = _pdfPlugin.get())
+        rect = plugin->convertFromPDFPageToScreenForAccessibility(bound, pageIndex);
+    return rect;
```
```suggestion
    if (RefPtr plugin = _pdfPlugin.get())
        return plugin->convertFromPDFPageToScreenForAccessibility(bound, pageIndex);
    return bound;
```

aprotyas reviewed (commented):
Left some comments but haven't done a full review yet. Will take a  look later today

aprotyas commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
```diff
@@ -120,6 +120,11 @@ class UnifiedPDFPlugin final : public PDFPluginBase, public WebCore::GraphicsLay
 
+#if PLATFORM(MAC)
+    WebCore::FloatRect convertFromPDFPageToScreenForAccessibility(const WebCore::FloatRect&, PDFDocumentLayout::PageIndex) const;
+    void accessibilityScrollToPage(int);
```
Can you address Simon's review comment here? This should be a `PDFDocumentLayout::PageIndex` instead of an `int`.

aprotyas commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
```diff
@@ -138,6 +138,12 @@ - (void)formChanged:(NSNotification *)notification
         m_annotationContainer->appendChild(annotationStyleElement);
         RefPtr { document->bodyOrFrameset() }->appendChild(*m_annotationContainer);
     }
```
Add a new line here please

aprotyas commented on Source/WebKit/WebProcess/Plugins/PDF/WKAccessibilityPDFDocumentObject.h:
```diff
@@ -41,19 +43,20 @@ class WeakPtrImplWithEventTargetData;
+- (NSObject *)accessibilityParent;
 - (id)accessibilityHitTest:(NSPoint)point;
 - (void)gotoDestination:(PDFDestination *)destination;
+- (NSRect)convertFromPDFPageToScreenForAccessibility:(NSRect)rectInPageCoordinate pageIndex:(WebKit::PDFDocumentLayout::PageIndex)pageIndex;
```
I don't think this is the selector you want to implement? Is anyone calling into this?

I see that PDFKit expects `convertFromPDFViewToScreenForAccessibility`

aprotyas commented on Source/WebKit/WebProcess/Plugins/PDF/WKAccessibilityPDFDocumentObject.mm:
```diff
@@ -101,13 +115,28 @@ - (NSArray *)accessibilityVisibleChildren
+    NSSize size = [[accessibilityParent accessibilityAttributeValue:NSAccessibilitySizeAttribute] sizeValue];
+    NSPoint origin = [[accessibilityParent accessibilityAttributeValue:NSAccessibilityPositionAttribute] pointValue];
+ALLOW_DEPRECATED_DECLARATIONS_END
+    return (NSRect) { origin, size };
```
optional nit: `return NSMakeRect(origin.x, origin.y, size.width, size.height);`

aprotyas replied:
We don't usually (ever?) construct the way you've written here.

aprotyas commented on Source/WebKit/WebProcess/Plugins/PDF/WKAccessibilityPDFDocumentObject.mm:
```diff
@@ -231,12 +263,10 @@ - (void)gotoDestination:(PDFDestination *)destination
-    NSInteger pageIndex = [_pdfDocument indexForPage:[destination page]];
-    NSRect rect = [[self accessibilityChildren][0] accessibilityFrame];
-    WebCore::IntPoint point = { 0, static_cast<int>(rect.size.height * pageIndex) + 1 };
+    int pageIndex = (int) [_pdfDocument indexForPage:[destination page]];
```
You can rewrite this with `PDFDocumentLayout::PageIndex`
```suggestion
    PDFDocumentLayout::PageIndex pageIndex = [_pdfDocument indexForPage:[destination page]];
```

aprotyas replied:
This will fit in nicely when you also let `UnifiedPDFPlugin::accessibilityScrollToPage` accept a `PDFDocumentLayout::PageIndex` argument.

aprotyas replied:
It seems like this suggestion didn't stick? Mind addressing this again?

```cpp
PDFDocumentLayout::PageIndex pageIndex = [_pdfDocument indexForPage:[destination page]];
callOnMainRunLoop([plugin, pageIndex] {
    plugin->accessibilityScrollToPage(pageIndex);
});
```

aprotyas reviewed (dismissed):
@ginalined most of the PR looks good, except for this method on your accessibility object:

```obj-c
- (NSRect)convertFromPDFPageToScreenForAccessibility:(NSRect)rectInPageCoordinate pageIndex:(WebKit::PDFDocumentLayout::PageIndex)pageIndex;
```

I think this is wrongly named and will never get called by PDFKit. Can you double check?

aprotyas reviewed (commented):
Looks great so far, just one more suggestion.

Also, can you please fix the commit message? The radar link is missing!


```
[UnifiedPDF] Fix coordinate space for accessibility.
https://bugs.webkit.org/show_bug.cgi?id=269956
rdar://123459892
```

aprotyas reviewed (approved):
Thanks for iterating! (r=mews)

aprotyas commented on Source/WebKit/WebProcess/Plugins/PDF/WKAccessibilityPDFDocumentObject.mm:
```diff
@@ -231,12 +263,10 @@ - (void)gotoDestination:(PDFDestination *)destination
-    NSInteger pageIndex = [_pdfDocument indexForPage:[destination page]];
-    NSRect rect = [[self accessibilityChildren][0] accessibilityFrame];
-    WebCore::IntPoint point = { 0, static_cast<int>(rect.size.height * pageIndex) + 1 };
+    PDFDocumentLayout::PageIndex pageIndex = [_pdfDocument indexForPage:[destination page]];
```
To fix the build:
```suggestion
    WebKit::PDFDocumentLayout::PageIndex pageIndex = [_pdfDocument indexForPage:[destination page]];
```

Commit:
[UnifiedPDF] Fix coordinate space for accessibility.

Reviewed by Abrar Rahman Protyasha.

- Fix VO cursor position does not match the focused element in unified pdf.
There are 2 parts of this change. One in pdfKit and one in webKit
- Update the plugin in WKAccessibilityDocumentObject to UnifiedPDFPlugin
- Remove extra functions
- Add new accessibility support api.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
(WebKit::PDFPluginBase::PDFPluginBase):
(WebKit::PDFPluginBase::convertFromPDFViewToRootView const): Deleted.
(WebKit::PDFPluginBase::convertFromPDFViewToScreenForAccessibility const): Deleted.
(WebKit::PDFPluginBase::convertFromRootViewToPDFView const): Deleted.
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::UnifiedPDFPlugin):
(WebKit::UnifiedPDFPlugin::accessibilityScrollToPage):
(WebKit::UnifiedPDFPlugin::convertFromPDFPageToScreenForAccessibility const):
(WebKit::UnifiedPDFPlugin::accessibilityHitTestIntPoint const):
* Source/WebKit/WebProcess/Plugins/PDF/WKAccessibilityPDFDocumentObject.h:
* Source/WebKit/WebProcess/Plugins/PDF/WKAccessibilityPDFDocumentObject.mm:
(-[WKAccessibilityPDFDocumentObject setPDFPlugin:]):
(-[WKAccessibilityPDFDocumentObject accessibilityFocusedUIElement]):
(-[WKAccessibilityPDFDocumentObject accessibilityWindow]):
(-[WKAccessibilityPDFDocumentObject accessibilityTopLevelUIElement]):
(-[WKAccessibilityPDFDocumentObject accessibilityVisibleChildren]):
(-[WKAccessibilityPDFDocumentObject accessibilitySubrole]):
(-[WKAccessibilityPDFDocumentObject accessibilityFrame]):
(-[WKAccessibilityPDFDocumentObject accessibilityParent]):
(-[WKAccessibilityPDFDocumentObject accessibilityAttributeValue:]):
(-[WKAccessibilityPDFDocumentObject accessibilityArrayAttributeCount:]):
(-[WKAccessibilityPDFDocumentObject accessibilityChildren]):
(-[WKAccessibilityPDFDocumentObject convertFromPDFPageToScreenForAccessibility:pageIndex:]):
(-[WKAccessibilityPDFDocumentObject accessibilityHitTest:]):
(-[WKAccessibilityPDFDocumentObject gotoDestination:]):
(-[WKAccessibilityPDFDocumentObject parent]): Deleted.

[UnifiedPDF] Fix coordinate space for accessibility https://bugs.webkit.org/show_bug.cgi?id=269956 rdar://123459892

Reviewed by Abrar Rahman Protyasha.

- Fix VO cursor position does not match the focused element in unified pdf.
There are 2 parts of this change. One in pdfKit and one in webKit
- Update the plugin in WKAccessibilityDocumentObject to UnifiedPDFPlugin
- Remove extra functions
- Add new accessibility support api

Canonical link: https://commits.webkit.org/275848@main
